### PR TITLE
First implementation of ResizableGrid and DockablePanel components

### PR DESCRIPTION
**Pull request by MV88**
This PR is for adding to the feature grid components two new elements.

The DockablePanel components wrap any the ResizableGrid 
Added tests and documentation

### CODE REVIEW

File: web/client/components/data/featuregrid/DockedGrid.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * @prop {number} maxDockSize. The maximum extension in % of the docked panel
+ * @prop {number} minDockSize. The minimum extension in % of the docked panel
+ * @prop {string} position the position of the docked panel
+ * @prop {function} setDockSize. The metod called when the docked panel is resized
+ *
+ */
+const DockedGrid = React.createClass({
+    propTypes: {
+        dockSize: React.PropTypes.number,
+        featureDockedGrid: React.PropTypes.oneOfType([React.PropTypes.object, React.PropTypes.func]),
```
Comment (mbarto) [inline web/client/components/data/featuregrid/DockedGrid.jsx]: grid

File: web/client/components/data/featuregrid/DockedGrid.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                isVisible={true}
+                onSizeChange={this.limitDockHeight}
+                fluid={true}
+                dimStyle={{ background: 'rgba(0, 0, 100, 0.2)' }}
+                dockStyle={null}
+                dockHiddenStyle={null}
+                 >
+                    <div id="feature-grid-container">
+                        <FeatureDockedGrid id="feature-docked-grid"
+                            toolbar={{
```
Comment (mbarto) [inline web/client/components/data/featuregrid/DockedGrid.jsx]: Can we configure this externally. We could have a grid without a toolbar

File: web/client/components/data/featuregrid/DockedGrid.jsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        position: React.PropTypes.string,
+        setDockSize: React.PropTypes.func
+    },
+    contextTypes: {
+        messages: React.PropTypes.object
+    },
+    getDefaultProps() {
+        return {
+            dockSize: 0.35,
+            featureDockedGrid: {},
```
Comment (offtherailz) [inline web/client/components/data/featuregrid/DockedGrid.jsx]: call it grid

File: web/client/components/data/featuregrid/DockedGrid.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                id="dock"
+                zIndex={1030 /*below dialogs, above left menu*/}
+                position={this.props.position /* 'left', 'top', 'right', 'bottom' */}
+                size={this.props.dockSize}
+                dimMode={"none" /*'transparent', 'none', 'opaque'*/}
+                isVisible={true}
+                onSizeChange={this.limitDockHeight}
+                fluid={true}
+                dimStyle={{ background: 'rgba(0, 0, 100, 0.2)' }}
+                dockStyle={null}
```
Comment (mbarto) [inline web/client/components/data/featuregrid/DockedGrid.jsx]: remove

File: web/client/components/data/featuregrid/DockedGrid.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                dockHiddenStyle={null}
```
Comment (mbarto) [inline web/client/components/data/featuregrid/DockedGrid.jsx]: remove

File: web/client/components/data/featuregrid/FeatureDockedGrid.jsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    },
+    getInitialState() {
+        return {
+            minHeight: this.props.minHeight
+        };
+    },
+    componentWillReceiveProps(newProps) {
+        if (this.props.size !== newProps.size) {
+            const pos = newProps.position;
+            if (pos === "right" || pos === "left") {
```
Comment (offtherailz) [inline web/client/components/data/featuregrid/FeatureDockedGrid.jsx]: props.autoWidth = true if pos === right or left
props.autoHeight idem

File: web/client/components/data/featuregrid/FeatureDockedGrid.jsx [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+ * @prop {string} rowKey the key used to distinguish rows
+ * @prop {object} rows. The features passed to the grid
+ * @prop {object} selectBy. It contains the selected rows
+ * @prop {number} size. The size of the dock panel wrapping this component
+ * @prop {object} style. The style of a wrapper div
+ * @prop {object} toolbar. The tools of the grid
+ * @prop {object} [tools] The tools of the grid
+ *
+ */
+const FeatureDockedGrid = React.createClass({
```
Comment (offtherailz) [inline web/client/components/data/featuregrid/FeatureDockedGrid.jsx]: Rename it ResisableGrid.

Comment (mbarto) [inline web/client/components/data/featuregrid/FeatureDockedGrid.jsx]: I don't see any Feature here

Comment (mbarto) [inline web/client/components/data/featuregrid/FeatureDockedGrid.jsx]: And no docked too

File: web/client/components/data/featuregrid/FeatureDockedGrid.jsx [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+        let tools = [];
+        if (this.props.toolbar.zoom) {
+            tools.push(<Button key="zoom" onClick={this.zoomToFeatures}><Glyphicon glyph="zoom-in"/></Button>);
+        }
+        const rowText = this.props.selectBy.keys.values.length === 1 ? 'row' : 'rows';
+        const rowsSelected = this.props.selectBy.keys.values.length || 0;
+        tools = [...tools, this.props.tools];
+
+        return (
+            <div style={{
```
Comment (offtherailz) [inline web/client/components/data/featuregrid/FeatureDockedGrid.jsx]: Wrap from the container.

File: web/client/components/data/featuregrid/FeatureDockedGrid.jsx [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+                        }}
+                    />
+                </div>
+                <div id="feature-grid-toolbar">
+                    <div style={{ position: "absolute",
+                        left: 2,
+                        bottom: 2}}>
+                        <span>{rowsSelected} {rowText} selected</span>
+                    </div>
+                    <ButtonToolbar className="featuregrid-tools" style={{marginLeft: "0px", flex: "none", marginBottom: "2px", marginRight: "2px"}} bsSize="sm">
```
Comment (offtherailz) [inline web/client/components/data/featuregrid/FeatureDockedGrid.jsx]: Add the toolbar in the container.

File: web/client/components/data/featuregrid/FeatureGrid.jsx [hunk 10]
```diff
@@ -188,7 +188,7 @@ const FeatureGrid = React.createClass({
                 flexDirection: "column",
                 height: "100%"
             }}>
-                <div fluid={false} style={this.props.style} className="ag-fresh">
```
Comment (offtherailz) [inline web/client/components/data/featuregrid/FeatureGrid.jsx]: Why this change? is related to the DockableGrid?

File: web/client/themes/default/ms2-theme.less [hunk 11]
```diff
@@ -952,15 +952,51 @@ select.form-control option {
...
+	font-family: @font-family-base;
+    height:55px !important;
+    text-align: center;
+    padding-top: 17px;
+}
+.react-grid-Row--even .react-grid-Cell { background-color: #f2f2f2; }
+
+.react-grid-Main, .react-grid-Container, #feature-grid-container {
+    height: 100%;
+    width: 100% !important;
```
Comment (offtherailz) [inline web/client/themes/default/ms2-theme.less]: avoid !important when possible. Try to provide instead inline style

File: web/client/themes/default/ms2-theme.less [hunk 12]
```diff
@@ -952,15 +952,51 @@ select.form-control option {
...
+}
+.react-grid-Container {
+    width: 100% !important;
```
Comment (offtherailz) [inline web/client/themes/default/ms2-theme.less]: avoid !important

File: web/client/themes/default/ms2-theme.less [hunk 13]
```diff
@@ -952,15 +952,51 @@ select.form-control option {
...
+}
+/*
+.react-grid-Canvas{
+    height: 100% !important;
+}*/
+.react-grid-Grid{
+    min-height: 100% !important;
```
Comment (offtherailz) [inline web/client/themes/default/ms2-theme.less]: avoid !important

File: web/client/components/data/featuregrid/FeatureGrid.jsx (hunk 10 above)
Comment (MV88) [inline web/client/components/data/featuregrid/FeatureGrid.jsx]: because fluid is not a parameter of the div tag

File: web/client/components/misc/DockablePanel.jsx [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+                dimMode={this.props.dimMode}
+                isVisible={this.props.isVisible}
+                onSizeChange={this.limitDockHeight}
+                fluid={this.props.fluid}
+                dimStyle={{ background: 'rgba(0, 0, 100, 0.2)' }}
+            >
+                <div id="container-wrapped-component" style={{height: "calc(100% - " + this.props.toolbarHeight + "px)"}}>
+                    {this.props.wrappedComponent !== null ? (<WrappedComponent
+                    size={{
+                        width: this.getAutoWidth(this.props.position),
```
Comment (offtherailz) [inline web/client/components/misc/DockablePanel.jsx]: use values instead of booleans and avoid the additional size parameter

File: web/client/components/misc/__tests__/ResizableGrid-test.jsx [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+    features: [{
+        id: "1",
+        name: "Edifici"
+    }, {
+        id: "2",
+        name: "Aiuola"
+    }, {
+        id: "3",
+        name: "Edifici"
+    }, {
```
Comment (offtherailz) [inline web/client/components/misc/__tests__/ResizableGrid-test.jsx]: less entries

File: web/client/components/misc/DockablePanel.jsx [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+                zIndex={this.props.zIndex}
+                position={this.props.position}
+                size={this.props.dockSize}
+                dimMode={this.props.dimMode}
+                isVisible={this.props.isVisible}
+                onSizeChange={this.limitDockHeight}
+                fluid={this.props.fluid}
+                dimStyle={{ background: 'rgba(0, 0, 100, 0.2)' }}
+            >
+                <div id="container-wrapped-component" style={{height: "calc(100% - " + this.props.toolbarHeight + "px)"}}>
```
Comment (offtherailz) [inline web/client/components/misc/DockablePanel.jsx]: use a class

File: web/client/themes/default/ms2-theme.less [hunk 17]
```diff
@@ -952,15 +952,65 @@ select.form-control option {
 	background-color: @ms2-color-primary !important;
 	color: @ms2-color-text-primary !important;
 }
-/* feauture grid */
+/* feature grid made by react-data-grid */
+#container-wrapped-component .react-grid-Header{
```
Comment (offtherailz) [inline web/client/themes/default/ms2-theme.less]: .container-wrapped-component (same below)
use a better name like dockpanel-wrapped-component or something like this

File: web/client/components/misc/DockablePanel.jsx [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+        toolbarHeight: React.PropTypes.number,
+        wrappedComponent: React.PropTypes.oneOfType([React.PropTypes.object, React.PropTypes.func]),
+        zIndex: React.PropTypes.number
+    },
+    contextTypes: {
+        messages: React.PropTypes.object
+    },
+    getDefaultProps() {
+        return {
+            id: "dock",
```
Comment (offtherailz) [inline web/client/components/misc/DockablePanel.jsx]: no default

File: web/client/components/misc/ResizableGrid.jsx [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+            <ReactDataGrid
+                columns={this.props.columns}
+                headerRowHeight={this.props.headerRowHeight}
+                minHeight={this.state.minHeight || this.props.minHeight}
+                minWidth={this.state.minWidth || this.props.minWidth}
+                ref={this.props.refGrid}
+                rowGetter={this.rowGetter}
+                rowHeight={this.props.rowHeight}
+                rowKey={this.props.rowKey}
+                rowSelection={{
```
Comment (offtherailz) [inline web/client/components/misc/ResizableGrid.jsx]: noSelection by default
